projects/plugins/jetpack/extensions/blocks/button/button-width-panel.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	BaseControl,
+	Button,
+	ButtonGroup,
+	PanelBody,
+	__experimentalUnitControl as UnitControl,
+} from '@wordpress/components';
+import { useEffect, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+const widthUnits = [
+	{ value: 'px', label: 'px', default: 150 },
+	{ value: '%', label: '%', default: 100 },
+	{ value: 'em', label: 'em', default: 10 },
+];
+
+const alignedWidthUnits = [
+	{ value: 'px', label: 'px', default: 150 },
+	{ value: 'em', label: 'em', default: 10 },
+];
+
+const predefinedWidths = [ '25%', '50%', '75%', '100%' ];
+
+export default function ButtonWidthPanel( props ) {
+	return (
+		<PanelBody title={ __( 'Width settings', 'jetpack' ) }>
+			<ButtonWidthControl { ...props } />
+		</PanelBody>
+	);
+}
+
+export function ButtonWidthControl( { align, width, onChange } ) {
+	const [ unit, setUnit ] = useState( null );
```

Comment:
The `unit` state gets defaulted to `null` here. This means that the `100` max for percentage widths isn't set on the `UnitControl` when a button with a percentage width is first loaded in the editor. You could parse the unit from the width value to use as the default instead.